### ISSUE TITLE
Not able to clear a text box in protractor

### ISSUE THREAD

**Issue by satishbit** [angular/protractor#4940]
Hi All,

I am using below Script to clear a Text Box in protractor. However the Text Box is not getting cleared. It gets appended with whatever i pass in SendKeys() function. Kindly help me on this.

Script used is as below

browser.element(by.xpath("")).clear();
browser.sleep(2000);
browser.element(by.xpath("")).sendKeys("100");

This Script does not works. Please help

**Comment by CrispusDH** [comment]
It would be better to create a question on the stack overflow. What exactly doesn't work? What error do you get?

**Comment by flekmatik** [comment]
We are having the same issue. Our suite started failing (maybe it's connected to chromedriver 2.43), input.clear() method just stopped working, no error message

**Comment by sergioalen** [comment]
> We are having the same issue. Our suite started failing (maybe it's connected to chromedriver 2.43), input.clear() method just stopped working, no error message

Check if you have a a `onblur` javascript execution. I've been having issues as well with clearing an input field which contains a validation function that runs onblur

**Comment by aarr0n** [comment]
Experiencing the same issue; `clear` method is just not working.

**Comment by wlsf82** [comment]
I'm also experiencing this issue with ChromeDriver 2.43, and downgrading to 2.42 fix the issue. :/

**Comment by CrispusDH** [comment]
You should add explicit waiter for checking that this element is clickable before use `clear`.

**Comment by kovacsbalu** [comment]
@CrispusDH waiting does not solve the onblur problem

**Comment by CrispusDH** [comment]
@kovacsbalu , you are right. In my case after updating to `Chrome 70` I have started to get errors before `clear` method. But before `clear` method I have already had explicit waiter on visibility of element. When I change it to `clickable` condition the errors gone. So, just in case...

**Comment by aarr0n** [comment]
Still failing with ChromeDriver 2.44.

**Comment by charlene17** [comment]
I've got the same issue, too. Rolling back to using ChromeDriver 2.42 works. But not working with ChromeDriver 2.43 +

**Comment by sergioalen** [comment]
> > > We are having the same issue. Our suite started failing (maybe it's connected to chromedriver 2.43), input.clear() method just stopped working, no error message
> > 
> > 
> > Check if you have a a `onblur` javascript execution. I've been having issues as well with clearing an input field which contains a validation function that runs onblur
> 
> @sergioalen Maybe I have the same. How did you fix this?
> `.....sendKeys(protractor.Key.SHIFT, protractor.Key.END, protractor.Key.BACK_SPACE, <data>`?

My issue was that before the update, I was clearing an input with `clear()` then entering a new value which with the new update `clear()` now focuses out the input field for some reason, triggering the `unblur` function. To stop this from happening I created a helper function to delete the values on the input field using backspace before entering the new value without focusing out the input field:

```
export function backspace(inputEl: ElementFinder, spaces?: number) {
  return inputEl.getAttribute('value').then((text) => {
    let spacesToGoBack = spaces || text.length;
    do {
      inputEl.sendKeys(Key.BACK_SPACE);
      spacesToGoBack--;
    } while (spacesToGoBack > 0);
  });
}
```
Then in the test I would use this like this:

```
backspace(inputEl).then(() => inputEl.sendKeys(price));
```

I'm not sure if that was the actual issue, but it worked for us.

**Comment by wlsf82** [comment]
I am facing the same issue, and in one specific case, even with chromedriver 2.42, when running the test in headless mode, the clear method does not work, but when running with headless mode Off, then it works.
It may be worth mentioning that I'm testing a React app, which uses `react-final-form` for the forms where I face this issue.

**Comment by praveendvd** [comment]
clear works only on the value attribute and not innerText , so we cannot use it to clear text field. Is there a work around ?

**Comment by praveendvd** [comment]
Try this to clear the field: 
  
   await elem.sendKeys( protractor.Key.chord(protractor.Key.CONTROL, "a"))
   await elem.sendKeys( protractor.Key.chord(protractor.Key.DEL))

**Comment by neekhu** [comment]
It works to delete text inside input element, if you use the following code:

element(by.id('Your_ID_Here')).sendKeys(protractor.Key.CONTROL,'a');
element(by.id('Your_ID_Here')).sendKeys(protractor.Key.BACK_SPACE;

**Comment by Fuun347** [comment]
If none of the above work, you can try this as well!
`let ctrlA = protractor.Key.chord(protractor.Key.CONTROL, "a"); 
await input.sendKeys(ctrlA);
await input.sendKeys(protractor.Key.BACK_SPACE);`

**Comment by danieldatri** [comment]
I was having the same problem with some inputs and I solved doing this method:

`
clearAndWriteField(fieldElement, text) {
    return fieldElement
      .getAttribute('value')
      .then((val) =>
        fieldElement
          .sendKeys(protractor.Key.ARROW_RIGHT.repeat(val.length))
          .sendKeys(protractor.Key.BACK_SPACE.repeat(val.length))
          .sendKeys(text)
      );
  }
`
